### ISSUE TITLE
Accessibility: message reading by screen readers is too verbose and not optimal

### ISSUE THREAD

**Issue by OSadovy** [mattermost/mattermost#30146]
There are multiple usability issues currently that make Mattermost less convenient for prolonged usageby screen reader users compared to slack and teams:
* The label that is generated and read when focusing a message while being "human sentence like" in practice is too verbose. Imagine reading tens or hundreds of messages and hearing "at (time) ... (date with weekday) ... (user) wrote," and only then hearing the actual message. this first part is very repetitive, it does not convey the most important informationthat user usually is looking to hear: namely who send the message and its text. The other information is useful but it is not primary and thus it should not go before the actual message content. If we put the most important bits first, the user might chose to interupt their screen reader by navigating to the next message before hearing the current date again and again if they just want to skim through the dialog. Both slack and teams do the user name and the message first. The date can go after the author, the message and meta information about attachments, replies, reactions etc.
* Because we hand craft the label text, markdown formatting is exposed directly to the user with all its special characters instead of the formatted text. e.g. Try to read some messages which contain links and you will quickly get what I mean. One easy fix for it is to use aria-labeledby with multiple ids, including the id of the div containing the message text, and then the browser engine and the screen reader will expose formatting by themselves.
* There currently is no indication of available replies for messages that start a thread. Also, there is no easy way to open and navigate into a thread via keyboard from the focused message. In slack, the user just presses right arrow while being on the starting message, thread opens and the first reply is focused. To close a thread and return to the first message, the user can press left arrow. This is very intuitive since it follows how treeviews work and users already know it.
* there is no easy way to get to the first unread message with keyboard navigation when opening a channel. ideally, it should be focused automatically when user presses shift+tab from the editor area.
* It should be possible to jump to the last/first message via the keyboard (by pressing home/end keys). the home key might bring you to either the first unread message or the previous date separator if all messages are read.
* there should be a way to fastforward/rewind messages via the keyboard (e.g. by pressing page up/page down) this could scroll the message view by X amount of messages.
* the date format that is exposed to screen reader could be improved to be less verbose. e.g. it is unnecessary to include the date if it is today, we can say "today" or skip the date entirely and just include the time. for yesterday we can say "yesterday" (as it is quicker to pronounce compared to the whole date). for dates earlier it should be fine to include them as is.

Overall, those improvements are pretty straightforward to implement, but they trully make a difference and show empathy to the users.

**Comment by amyblais** [comment]
Hi @OSadovy! What Mattermost server version are you using?

**Comment by OSadovy** [comment]
@amyblais this is more a client side concern, but you can reproduce by connecting to the public community mattermost server:
```
Mattermost Version: 10.6.0
Database Schema Version: 130
Build Number: 13233659518
Database: postgres
Hostname: mm-rxoc-7568cf5768-9zg22
```
the windows desktop client version is 5.10.2. The web client version shows the same behavior.

**Comment by amyblais** [comment]
Opened https://mattermost.atlassian.net/browse/MM-62934.

**Comment by rmwilliams2023** [comment]
I agree with all of the above in this issue, and it would be great to see some work towards improving these matters soon.

**Comment by asaadmahmood** [comment]
Hey @rmwilliams2023 @OSadovy Let me raise this directly with the team and get this prioritized.
